models/classes/UniqueId/Listener/TestCreationListener.php

```diff
@@ -78,14 +74,17 @@ public function populateUniqueId(Event $event): void
 
         $identifier = $this->identifierGenerator->generate([IdentifierGeneratorInterface::OPTION_RESOURCE => $test]);
 
-        $test->editPropertyValues(
-            $this->ontology->getProperty(TaoOntology::PROPERTY_UNIQUE_IDENTIFIER),
-            $identifier
-        );
         $this->qtiIdentifierSetter->set([
             AbstractQtiIdentifierSetter::OPTION_RESOURCE => $test,
             AbstractQtiIdentifierSetter::OPTION_IDENTIFIER => $identifier,
         ]);
+
+        if ($this->featureFlagChecker->isEnabled('FEATURE_FLAG_UNIQUE_NUMERIC_QTI_IDENTIFIER')) {
```

Comment:
@shpran Can you please validate that with test translations and `FEATURE_FLAG_UNIQUE_NUMERIC_QTI_IDENTIFIER` disabled it will not be an issue? The translated tests must have the identifier repeated either way, no?

Reply:
Good catch! Thanks
Covered it here: https://github.com/oat-sa/tao-core/pull/4206

Cc: @Nevermind23